Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -119,18 +133,45 @@ public bool TryGetXComponentCorrelationId(string instrumentationKey, out string
                             {
                                 this.GenerateCorrelationIdAndAddToDictionary(instrumentationKey, appId.Result);
                             }
+                            catch (AggregateException ex)
+                            {
+                                ex.Flatten();
```

Comment:
ex.Flatten() wouldn't flatten the exception in place. It will return a new flattened result.